compiler/semtypes.nim

```diff
@@ -1133,6 +1133,10 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
 
       if typ == nil:
         typ = def.typ
+        if typ.kind == tySequence and typ[0].kind == tyEmpty or
```

Comment:
Instead use ``isEmptyContainer(typ)``

Reply:
Nice helper method, forget about sets. This also returns 'empty' on nil so this awful code would stop working.
```nim
type
   SomeRef = ref object
      val: int
proc foo(a: SomeRef) = echo a.val
proc bar(a = nil) = a.foo()
bar() # segfaults of course
```
IMO that working is a bug. But is that a bug?

Reply:
``nil`` is a type of its own but it's fine to disallow ``a = nil`` in parameter lists. Btw I think ``testament/tester cat megatest`` fails with your PR.

Reply:
Can't reproduce failure locally. The time the tests took don't seem to add up to 90 mins, so idk why it timed out.